Source/Dafny/Compilers/Compiler.cs

```diff
@@ -4736,6 +4740,53 @@ private class GetterSetterLvalueImpl : ILvalue {
       }
     }
 
+    private ConcreteSyntaxTree MaybeInjectSubsetConstraint(ConcreteSyntaxTree wr, bool inLetExprBody, BoundVar bv,
+      Expression e, bool isReturning = false, bool elseReturnValue = false) {
+      int i;
+      if (bv.Type is UserDefinedType
+        {
+          TypeArgs: var typeArgs,
+          ResolvedClass:
+        var subsetTypeDecl and
+        SubsetTypeDecl
```

Comment:
C# has this `var` pattern to assign a field to a local variable, and the `Class { ... }` to do extractions. To merge both, it defines the "and" pattern builder that can do the two. By the way, here `subsetTypeDecl ` has type `TopLevelDecl`, but if I reversed the operands, it would have the type `SubsetTypeDecl`.
In any case, I'll remove it because it is not needed here anymore. I extracted all relevant fields.